test/base/test_history.py

```diff
@@ -689,6 +691,57 @@ def test_hdf5_history_mp():
                             )
 
 
+def test_hdf5_amici_history():
+    objective1 = pypesto.Objective(
+        fun=so.rosen, grad=so.rosen_der, hess=so.rosen_hess
+    )
+    objective2 = load_amici_objective('conversion_reaction')[0]
+    lb = -2 * np.ones((1, 2))
+    ub = 2 * np.ones((1, 2))
+    problem1 = pypesto.Problem(objective=objective1, lb=lb, ub=ub)
+    problem2 = pypesto.Problem(objective=objective2, lb=lb, ub=ub)
+
+    optimizer = pypesto.optimize.ScipyOptimizer(options={'maxiter': 10})
+
+    with tempfile.TemporaryDirectory(dir=".") as tmpdirname:
+        for f_ext, amici_history_class in zip(
+            [".csv", ".hdf5"], [CsvAmiciHistory, Hdf5AmiciHistory]
+        ):
+            _, fn = tempfile.mkstemp(f_ext, '{id}', dir=f"{tmpdirname}")
+
+            history_options_mp = pypesto.HistoryOptions(
+                trace_record=True, storage_file=fn
+            )
+
+            result1 = pypesto.optimize.minimize(
+                problem=problem1,
+                optimizer=optimizer,
+                n_starts=1,
+                history_options=history_options_mp,
+                progress_bar=False,
+            )
+            assert not isinstance(
+                result1.optimize_result.list[0].history, amici_history_class
+            )
+
+            # optimizing with amici history saved in hdf5
+            result2 = pypesto.optimize.minimize(
```

Comment:
`ValueError: ID "0" is already used in history file`

-> delete file before second minimize or use a different file